codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttMessageBuilder.java

```diff
@@ -0,0 +1,259 @@
+        private String password;
+
+        public ConnectBuilder protocolVersion(MqttVersion version) {
+            this.version = version;
+            return this;
+        }
+
+        public ConnectBuilder clientId(String clientId) {
+            this.clientId = clientId;
+            return this;
+        }
+
+        public ConnectBuilder cleanSession(boolean cleanSession) {
+            this.cleanSession = cleanSession;
+            return this;
+        }
+
+        public ConnectBuilder keepAlive(int keepAliveSecs) {
+            this.keepAliveSecs = keepAliveSecs;
+            return this;
+        }
+
+        public ConnectBuilder willFlag(boolean willFlag) {
+            this.willFlag = willFlag;
+            return this;
+        }
+
+        public ConnectBuilder willQoS(MqttQoS willQos) {
+            this.willQos = willQos;
+            return this;
+        }
+
+        public ConnectBuilder willTopic(String willTopic) {
+            this.willTopic = willTopic;
+            return this;
+        }
+
+        public ConnectBuilder willMessage(String willMessage) {
+            this.willMessage = willMessage;
+            return this;
+        }
+
+        public ConnectBuilder willRetain(boolean willRetain) {
+            this.willRetain = willRetain;
+            return this;
+        }
+
+        public ConnectBuilder hasUser() {
+            this.hasUser = true;
+            return this;
+        }
+
+        public ConnectBuilder hasPassword() {
+            this.hasPassword = true;
+            return this;
+        }
+
+        public ConnectBuilder username(String username) {
+            this.username = username;
+            return this;
+        }
+
+        public ConnectBuilder password(String password) {
+            this.password = password;
+            return this;
+        }
+
+        public MqttConnectMessage build() {
+            MqttFixedHeader mqttFixedHeader =
+                    new MqttFixedHeader(MqttMessageType.CONNECT, false, MqttQoS.AT_MOST_ONCE, false, 0);
+            MqttConnectVariableHeader mqttConnectVariableHeader =
+                    new MqttConnectVariableHeader(
+                            version.protocolName(),
+                            version.protocolLevel(),
+                            hasUser,
+                            hasPassword,
+                            willRetain,
+                            willQos.value(),
+                            willFlag,
+                            cleanSession,
+                            keepAliveSecs);
+            MqttConnectPayload mqttConnectPayload =
+                    new MqttConnectPayload(clientId, willTopic, willMessage, username, password);
+            return new MqttConnectMessage(mqttFixedHeader, mqttConnectVariableHeader, mqttConnectPayload);
+        }
+    }
+
+    public static class SubscribeBuilder {
+
+        private List<MqttTopicSubscription> subscriptions = new ArrayList<MqttTopicSubscription>();
```

Comment:
consider lazy instance it to eliminate unessary allocations when no subscriptions are added.